dpppt-backend-sdk/dpppt-backend-sdk-ws/src/test/java/org/dpppt/backend/sdk/ws/controller/GaenV2ControllerTest.java

```diff
@@ -97,6 +97,63 @@ public void testNoTokenFails() throws Exception {
     assertTrue(authenticateError.contains("Bearer"));
   }
 
+  private void testJWTFakeNoKeysInserted(boolean useV2Upload) throws Exception {
+    var now = UTCInstant.now();
+    List<GaenKey> keys = new ArrayList<>();
+    for (int i = 0; i < 30; i++) {
+      var tmpKey = new GaenKey();
+      tmpKey.setRollingStartNumber((int) now.atStartOfDay().minusDays(i).get10MinutesSince1970());
+      var keyData = String.format("testKey32Bytes%02d", i);
+      tmpKey.setKeyData(Base64.getEncoder().encodeToString(keyData.getBytes("UTF-8")));
+      tmpKey.setRollingPeriod(144);
+      tmpKey.setFake(0);
+      tmpKey.setTransmissionRiskLevel(0);
+      keys.add(tmpKey);
+    }
+
+    Object uploadPayload;
+    if (useV2Upload) {
+      GaenV2UploadKeysRequest exposeeRequest = new GaenV2UploadKeysRequest();
+      exposeeRequest.setGaenKeys(keys);
+      uploadPayload = exposeeRequest;
+    } else {
+      GaenRequest exposeeRequest = new GaenRequest();
+      exposeeRequest.setGaenKeys(keys);
+      exposeeRequest.setDelayedKeyDate((int) now.atStartOfDay().get10MinutesSince1970());
+      uploadPayload = exposeeRequest;
+    }
+
+    String token = createToken(true, now.plusMinutes(5));
```

Comment:
As in the test below, I would run the test twice: once with `createToken(false, ...` and once with `createToken(true`, just to make sure you're not missing some keys due to another part of the logic.

Even though I know some of the code, I couldn't guarantee that this actually passes with `createToken(false`...